yarn-project/world-state/src/synchroniser/server_world_state_synchroniser.ts

```diff
@@ -92,6 +113,7 @@ export class ServerWorldStateSynchroniser implements WorldStateSynchroniser {
     await this.jobQueue.cancel();
     await this.merkleTreeDb.stop();
     await this.runningPromise;
+    await this.commitCurrentL2BlockNumber();
```

Comment:
Can we do this at the end of collectAndProcessBlocks()? If we don't have a clean shutdown then it's likely that the stored block number will be out of sync with the trees.